applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -1491,93 +1156,141 @@ static int initialize(le_audio_receive_cb recv_cb, le_audio_timestamp_cb timestm
+	return -ENOTSUP;
 }
 
 int le_audio_config_get(uint32_t *bitrate, uint32_t *sampling_rate, uint32_t *pres_delay)
 {
-	LOG_WRN("Getting config from gateway is not yet supported");
+	ARG_UNUSED(bitrate);
+	ARG_UNUSED(sampling_rate);
+	ARG_UNUSED(pres_delay);
+
+	LOG_WRN("%s not supported", __func__);
 	return -ENOTSUP;
 }
 
-int le_audio_volume_up(void)
+void le_audio_conn_set(struct bt_conn *conn)
 {
 	int ret;
 
-	ret = ble_vcs_volume_up();
+	ret = discover_sink(conn);
 	if (ret) {
-		LOG_WRN("Failed to increase volume");
-		return ret;
+		LOG_WRN("Failed to discover sink: %d", ret);
 	}
-
-	return 0;
 }
 
-int le_audio_volume_down(void)
+void le_audio_conn_disconnected(struct bt_conn *conn)
 {
 	int ret;
+	uint8_t channel_index;
 
-	ret = ble_vcs_volume_down();
+	ret = channel_index_get(conn, &channel_index);
 	if (ret) {
-		LOG_WRN("Failed to decrease volume");
-		return ret;
+		LOG_WRN("Unknown connection disconnected");
+	} else {
+		disconnected_headset_cleanup(channel_index);
 	}
+}
 
-	return 0;
+int le_audio_ext_adv_set(struct bt_le_ext_adv *ext_adv)
+{
+	ARG_UNUSED(ext_adv);
+
+	LOG_WRN("%s not supported", __func__);
+	return -ENOTSUP;
 }
 
-int le_audio_volume_mute(void)
+void le_audio_adv_get(const struct bt_data **adv, size_t *adv_size, bool periodic)
 {
-	int ret;
+	ARG_UNUSED(adv);
+	ARG_UNUSED(adv_size);
+	ARG_UNUSED(periodic);
 
-	ret = ble_vcs_volume_mute();
-	if (ret) {
-		LOG_WRN("Failed to mute volume");
-		return ret;
+	LOG_WRN("%s not supported", __func__);
+}
+
+int le_audio_pa_sync_set(struct bt_le_per_adv_sync *pa_sync, uint32_t broadcast_id)
+{
+	ARG_UNUSED(pa_sync);
+	ARG_UNUSED(broadcast_id);
+
+	LOG_WRN("%s not supported", __func__);
+	return -ENOTSUP;
+}
+
+int le_audio_play(void)
+{
+	int ret_left = 0;
+	int ret_right = 0;
+
+	if (ep_state_check(headsets[AUDIO_CH_L].sink_stream.ep, BT_BAP_EP_STATE_QOS_CONFIGURED)) {
+		ret_left = bt_bap_stream_enable(&headsets[AUDIO_CH_L].sink_stream,
+						lc3_preset_sink.codec.meta,
+						lc3_preset_sink.codec.meta_count);
+
+		if (ret_left) {
+			LOG_WRN("Failed to enable left stream: %d", ret_left);
+		}
 	}
 
+	if (ep_state_check(headsets[AUDIO_CH_R].sink_stream.ep, BT_BAP_EP_STATE_QOS_CONFIGURED)) {
+		ret_right = bt_bap_stream_enable(&headsets[AUDIO_CH_R].sink_stream,
+						 lc3_preset_sink.codec.meta,
+						 lc3_preset_sink.codec.meta_count);
+
+		if (ret_right) {
+			LOG_WRN("Failed to enable right stream: %d", ret_right);
+		}
+	}
+
+	if (ret_left || ret_right) {
+		return -EIO;
+	}
+
+	playing_state = true;
+
 	return 0;
 }
 
-int le_audio_play_pause(void)
+int le_audio_pause(void)
 {
-	int ret;
+	int ret_left = 0;
+	int ret_right = 0;
 
-	if (IS_ENABLED(CONFIG_STREAM_BIDIRECTIONAL)) {
-		LOG_WRN("Play/pause not supported for bidirectional mode");
-	} else {
-		ret = ble_mcs_play_pause(NULL);
-		if (ret) {
-			LOG_WRN("Failed to change streaming state");
-			return ret;
+	le_audio_event_publish(LE_AUDIO_EVT_NOT_STREAMING, NULL);
```

Comment:
Should the event be published before the stream has been disabled?

Reply:
Yes, because we want to stop the audio system before the ISO stream is disabled, otherwise we will get a lot warning stating that audio data can not be sent